core/src/test/java/io/grpc/internal/AbstractClientStreamTest.java

```diff
@@ -0,0 +1,254 @@
+package io.grpc.internal;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import io.grpc.ChannelImpl;
+import io.grpc.MessageEncoding;
+import io.grpc.Metadata;
+import io.grpc.Metadata.Headers;
+import io.grpc.Status;
+import io.grpc.Status.Code;
+import io.grpc.internal.AbstractStream.Phase;
+import io.grpc.internal.MessageFramerTest.ByteWritableBuffer;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.InputStream;
+
+/**
+ * Test for {@link AbstractClientStream}.  This class tries to test functionality in
+ * AbstractClientStream, but not in any super classes.
+ */
+public class AbstractClientStreamTest {
+
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+
+  @Mock private ClientStreamListener mockListener;
+  @Captor private ArgumentCaptor<Status> statusCaptor;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  private final WritableBufferAllocator allocator = new WritableBufferAllocator() {
+    @Override
+    public WritableBuffer allocate(int capacityHint) {
+      return new ByteWritableBuffer(capacityHint);
+    }
+  };
+
+  @Test
+  public void cancel_failsOnUnsupportedCode() {
```

Comment:
nit: I think the recommended naming would be something like `cancelShouldFailOnUnsupportedCode`.